cas-server-support-couchbase-core/src/main/java/org/jasig/cas/couchbase/core/CouchbaseClientFactory.java

```diff
@@ -0,0 +1,154 @@
+package org.jasig.cas.couchbase.core;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.couchbase.client.java.Bucket;
+import com.couchbase.client.java.Cluster;
+import com.couchbase.client.java.CouchbaseCluster;
+import com.couchbase.client.java.view.DesignDocument;
+import com.couchbase.client.java.view.View;
+
+/**
+ * A factory class which produces a client for a particular Couchbase bucket.
+ * A design consideration was that we want the server to start even if Couchbase
+ * is unavailable, picking up the connection when Couchbase comes online. Hence
+ * the creation of the client is made using a scheduled task which is repeated
+ * until successful connection is made.
+ *
+ * @author Fredrik Jönsson "fjo@kth.se"
+ * @author Misagh Moayyed
+ * @since 4.2
+ */
+public class CouchbaseClientFactory extends TimerTask {
+    private static final int RETRY_INTERVAL = 10; // seconds.
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final Timer timer = new Timer();
+
+    private Cluster cluster;
+    private Bucket bucket;
+    private List<View> views;
+
+    @NotNull
+    private List<String> nodes;
+
+    /* The name of the bucket, will use the default bucket unless otherwise specified. */
+    private String bucketName = "default";
+
+    /* Password for the bucket if any. */
+    private String password = "";
+
+    /* Design document and views to create in the bucket, if any. */
+    private String designDocument;
+
+    /**
+     * Instantiates a new Couchbase client factory.
+     */
+    public CouchbaseClientFactory() {}
+
+    /**
+     * Start initializing the client. This will schedule a task that retries
+     * connection until successful.
+     */
+    public void initialize() {
```

Comment:
This is called by either the service/ticket registry components @PostConstruct methods.